### PR TITLE
[1.2.0] Trailing whitespace in dhcpd.subnet.conf.j2

### PR DESCRIPTION
**Pull request by hmeAtos**
Hello,
When we define several networks (`ice1-2`, `ice1-3`, ...) , the `dhcpd.subnet.conf.j2` template generate many trailing whitespace in DHPC configuration files.
This is not a problem to run the DHCP service, but the configuration files generate is too big and not very clean.

For fix this issue, please merge my patch.

Regards,

### CODE REVIEW

File: roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2 [hunk 1]
```diff
@@ -37,14 +37,15 @@
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_' + ipxe_driver + 'ipxe.efi') %}
     {% else %}  {# Assume everything else is pcbios #}
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_undionly.kpxe') %}
-    {% endif %}
+    {% endif -%}
 
-    {% for nic, nic_args in hostvars[host]['network_interfaces'].items() %}
+    {%- for nic, nic_args in hostvars[host]['network_interfaces'].items() %}
```
Comment (actatux) [inline roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2]: I'm still not sure this minus sign at the beginning of the block changes anything in the output.
Can you confirm the output file is different without this change?

Comment (hmeAtos) [reply roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2]: Dear @actatux,
I will trying without this minus sign.
I'll keep you informed.
Regards

Comment (hmeAtos) [reply roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2]: Hello @actatux,
The minus sign adding causes change.
Regards

**Comment by oxedions** [comment]
@actatux Can you confirm your approval on this one ?

File: roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2 (hunk 1 above)
Comment (actatux) [reply roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2]: Thank you @hmeAtos. I was curious about the change, so I ran a test on top of this PR, with and without the minus sign at lines 42 and 82.
The main changes are the order of the hosts in the files and a few trailing whitespaces moving from one line to the other.

I can reproduce the same behaviour without modifying your PR, after 2 consecutive runs.

```
$ md5sum /tmp/dhcpd.subnet.conf.?
b890fecbb81697e7a04bbca02d555bbe  dhcpd.subnet.conf.1
9311154d685103f81048a203cd640742  dhcpd.subnet.conf.2
$ grep -c '^ \+$' dhcpd.subnet.conf.?
4
```

You can remove the trailing whitespaces with either:
```diff
--- a/roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2
+++ b/roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2
@@ -20,12 +20,12 @@
       {% set ipxe_driver = 'snponly_' %}
     {% elif hostvars[host]['equipment_profile']['ipxe_driver'] == 'snp' %}
       {% set ipxe_driver = 'snp_' %}
-    {% else %}  {# Assume everything else is default driver #}
+    {% else %}{# Assume everything else is default driver #}
       {% set ipxe_driver = '' %}
     {% endif %}
     {% if hostvars[host]['equipment_profile']['hardware']['cpu']['architecture'] == 'arm64' %} {# Now select architecture #}
       {% set ipxe_architecture = 'arm64' %}
-    {% else %}  {# Assume everything else is x86_64 #}
+    {% else %}{# Assume everything else is x86_64 #}
       {% set ipxe_architecture = 'x86_64' %}
     {% endif %}
     {% if hostvars[host]['equipment_profile']['ipxe_embed'] == 'dhcpretry' %} {# Now select embed script #}
@@ -35,7 +35,7 @@
     {% endif %}
     {% if hostvars[host]['equipment_profile']['ipxe_platform'] == 'efi' %} {# Finally, build file name depending of platform and previous parameters #}
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_' + ipxe_driver + 'ipxe.efi') %}
-    {% else %}  {# Assume everything else is pcbios #}
+    {% else %}{# Assume everything else is pcbios #}
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_undionly.kpxe') %}
     {% endif -%}
```
or
```diff
--- a/roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2
+++ b/roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2
@@ -20,12 +20,12 @@
       {% set ipxe_driver = 'snponly_' %}
     {% elif hostvars[host]['equipment_profile']['ipxe_driver'] == 'snp' %}
       {% set ipxe_driver = 'snp_' %}
-    {% else %}  {# Assume everything else is default driver #}
+    {% else %}  {#- Assume everything else is default driver #}
       {% set ipxe_driver = '' %}
     {% endif %}
     {% if hostvars[host]['equipment_profile']['hardware']['cpu']['architecture'] == 'arm64' %} {# Now select architecture #}
       {% set ipxe_architecture = 'arm64' %}
-    {% else %}  {# Assume everything else is x86_64 #}
+    {% else %}  {#- Assume everything else is x86_64 #}
       {% set ipxe_architecture = 'x86_64' %}
     {% endif %}
     {% if hostvars[host]['equipment_profile']['ipxe_embed'] == 'dhcpretry' %} {# Now select embed script #}
@@ -35,7 +35,7 @@
     {% endif %}
     {% if hostvars[host]['equipment_profile']['ipxe_platform'] == 'efi' %} {# Finally, build file name depending of platform and previous parameters #}
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_' + ipxe_driver + 'ipxe.efi') %}
-    {% else %}  {# Assume everything else is pcbios #}
+    {% else %}  {#- Assume everything else is pcbios #}
       {% set ipxe_rom_filename = (ipxe_architecture + '/' + ipxe_embed + '_undionly.kpxe') %}
     {% endif -%}
```

I also notice an alignment issue:
```
  host lawfer {
    option host-name "lawfer";
    hardware ethernet 08:00:27:0d:f8:a3;
    fixed-address 10.13.3.2;
    filename "x86_64/standard_undionly.kpxe";
  }

    host blawfer {
      option host-name "blawfer";
      hardware ethernet 08:00:27:0d:f8:b3;
      fixed-address 10.13.103.2;
    }
```

Comment (actatux) [reply roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2]: Hi @hmeAtos ,
Could you find some time to apply above recommendation next week?

**Comment by oxedions** [comment]
Dear @hmeAtos 

Could you also correct roles/core/dhcp_server/templates/dhcpd.subnet.conf.j2 as suggested by @actatux ?

So I can merge this :-)

Ben

File: roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2 (hunk 1 above)
Comment (hmeAtos) [reply roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2]: Hi @actatux and @oxedions,
I have apply this recommendation here : [PR:133](https://github.com/bluebanquise/bluebanquise/pull/133)

Indeed, these trailling spaces have caused problems. Now, they are fixed.
I suggest you apply my two PRs to fix the role advanced_dhcp_server.
Regards,

**Comment by hmeAtos** [comment]
> Dear @hmeAtos
> 
> Could you also correct roles/core/dhcp_server/templates/dhcpd.subnet.conf.j2 as suggested by @actatux ?
> 
> So I can merge this :-)
> 
> Ben

Done.

File: roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2 (hunk 1 above)
Comment (actatux) [reply roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2]: Merging one will probably need to rebase the other. Stay tuned!

**Comment by actatux** [comment]
> > Dear @hmeAtos
> > Could you also correct roles/core/dhcp_server/templates/dhcpd.subnet.conf.j2 as suggested by @actatux ?
> > So I can merge this :-)
> > Ben
> 
> Done.

Where?

File: roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2 (hunk 1 above)
Comment (hmeAtos) [reply roles/advanced-core/advanced_dhcp_server/templates/dhcpd.subnet.conf.j2]: > > > Dear @hmeAtos
> > > Could you also correct roles/core/dhcp_server/templates/dhcpd.subnet.conf.j2 as suggested by @actatux ?
> > > So I can merge this :-)
> > > Ben
> > 
> > 
> > Done.
> 
> Where?

here: https://github.com/bluebanquise/bluebanquise/pull/133/commits/08170505dae4cb0471f6ddd7bb10b7ad2bce95c8

**Comment by actatux** [comment]
Hello @hmeAtos ,

We are talking about the core role that has the same issue than the advanced-core role. Could you add the same changes to this PR to solve the same issue in a single PR?

Thanks!

**Comment by hmeAtos** [comment]
> Hello @hmeAtos ,
> 
> We are talking about the core role that has the same issue than the advanced-core role. Could you add the same changes to this PR to solve the same issue in a single PR?
> 
> Thanks!

My bad, sorry, i did'nt understand.
It is commited here: [33eb38c](https://github.com/bluebanquise/bluebanquise/pull/107/commits/33eb38c37f63c4922533d14926411457931d0c27)

**Pull request merged**